plugin/transfer/transfer.go

```diff
@@ -0,0 +1,176 @@
+package transfer
+
+import (
+	"context"
+	"errors"
+
+	"github.com/coredns/coredns/plugin"
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/coredns/coredns/request"
+
+	"github.com/miekg/dns"
+)
+
+var log = clog.NewWithPlugin("transfer")
+
+// Transfer is a plugin that handles zone transfers.
+type Transfer struct {
+	xfrs []*xfr
+	Next plugin.Handler
+}
+
+type xfr struct {
+	Zones       []string
+	to          []string
+	Transferers []Transferer // the list of plugins that implement Transferer
```

Comment:
Why is this here instead of in the `Transfer` struct? In the setup you just add them all to every one anyway.

Reply:
ah, yes - it was there because it used to be a map of zones to Transferers ... only the zones for that instance were in it.  It can be moved to the main struct now.

Reply:
The list of Transferers is not likely to be large, and the load of transfer requests not likely to be high.  So the map optimization was not really needed.  Code is simpler without it.